Merge upstream to master

pastakhov opened the pull request.

vedmaka committed:
Upgrade sentry/sentry to 0.20.0

As from  what I did experience the extension had issues working
with old library versions (that's why the upgrade was needed) but
also it was unable to correctly work with latest or newer versions
of the library. I did pick 0.20.0 as a working stable version,
but tbh I did not test more recent 0.2x

Not sure if that's really needed since this patch can be considered
more like a workaround and apparently the extension as a whole requires
an upgrade to be compatible with latest library version and migrate away
from using legacy Sentry SDK

Change-Id: I97ff68be118e5b067e0dde044fd50de03b9a9ed2

reedy committed:
Update mediawiki/mediawiki-phan-config to 0.11.1

Bug: T295285
Change-Id: Iecbf9f6afc56205d29f9bd0b2a012fcade36d8cb

Commit:
build: Updating dependencies

composer:
* mediawiki/mediawiki-codesniffer: 38.0.0 → 39.0.0
* mediawiki/mediawiki-phan-config: 0.11.1 → 0.12.0
* php-parallel-lint/php-console-highlighter: 0.5.0 → 1.0.0
* php-parallel-lint/php-parallel-lint: 1.3.1 → 1.3.2

npm:
* grunt: 1.4.0 → 1.5.3
* grunt-banana-checker: 0.9.0 → 0.10.0
* grunt-eslint: 23.0.0 → 24.0.0
* async: 3.2.0 → 3.2.4
  * https://github.com/advisories/GHSA-fwr7-v2mv-hh25
* minimatch: 3.0.4 → 3.0.8
  * https://github.com/advisories/GHSA-f8q6-p94x-37v3

Additional changes:
* Set `name` in package.json.

Change-Id: Ib31fcc633f57bb5033331d1b5e2b796a59f6701c

translatewiki committed:
Localisation updates from https://translatewiki.net.

Change-Id: I8150e894e7e8fc70dddddfcd1f5d8c404d07f236

Commit:
build: Updating mediawiki/mediawiki-codesniffer to 40.0.1

Change-Id: I1ed5caadc42f16f59d28dc63f063304ee16ebc34

pastakhov commented on composer.json:
```diff
@@ -11,14 +11,14 @@
 	"require": {
 		"composer/installers": ">=1.0.1",
-		"sentry/sentry": "0.12.1"
+		"sentry/sentry": "0.20.0"
```
it moves me crazy, why cannot I see real difference between the branches?   it is already "0.20.0" in our master branch

vedmaka replied:
hmmm .. weird, either way, we're on 3.8.x version already so we can't sync with the wikimedia repo

translatewiki committed:
Localisation updates from https://translatewiki.net.

Change-Id: I02f6efea2fe564c261451d22bac91894a6f99f19

umherirrender committed:
Allow composer/installers plugin

`composer update --no-interactive` errors out if this flag is not set.
Setting it to true, as that was the status quo before Composer 2.2.

Bug: T298283
Change-Id: Ie7ce73e9a70b87a9aee0f6dee782f389d5b422f0

umherirrender committed:
build: Updating composer dependencies

* mediawiki/mediawiki-codesniffer: 40.0.1 → 41.0.0

Change-Id: Ie05195789012d847b83d47b7b0c1313bfe66dbfb

umherirrender committed:
tests/qunit: Change assert.ok to assert.true

Avoid loose assertion

Change-Id: I949d35aae71a5d69f4e5498ea57fc77407f9573d

Commit:
build: Updating npm dependencies

* eslint-config-wikimedia: 0.20.0 → 0.24.0
* grunt: 1.5.3 → 1.6.1

Change-Id: I05662a4e5e0dcbcf2cffa1ac3427a4f59a79e385

umherirrender committed:
Handle null $wgSentryDsn

PHP Deprecated: strpos(): Passing null to parameter #1 ($haystack) of
type string is deprecated in php8.1

Change-Id: I628d8cd3ed5497e6826e2f26f6038efd6c2a9dad

umherirrender committed:
Fix handling of null $wgSentryDsn

Follow-up: I628d8cd3ed5497e6826e2f26f6038efd6c2a9dad
Change-Id: Ic0b56548d33f7bbf519be169e12b6e34137db6df

Commit:
build: Updating dependencies

composer:
* mediawiki/mediawiki-phan-config: 0.12.0 → 0.12.1

npm:
* grunt-eslint: 24.0.0 → 24.0.1

Change-Id: Iedd6aa04fd95c1277d02600a55f9d1a2efc05263

Commit:
build: Updating eslint-config-wikimedia to 0.25.0

Change-Id: Iecdbd217e1ff02f99ad7885351ac6e0ab6459435

Commit:
build: Updating grunt-banana-checker to 0.11.0

Change-Id: I6b0b9c03292d64ac59c2e4664d33d821846e7540

vedmaka reviewed (commented):
we can't get this merged due to dependency on a much higher version of `sentry/sentry`, I propose to close the PR

Commit:
build: Updating npm dependencies

* eslint-plugin-compat: 4.1.2 → 4.2.0
  * https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
  * https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
  * https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
* semver: 7.3.8 → 7.5.4
  * https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
  * https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
  * https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
* word-wrap: 1.2.3 → 1.2.5
  * https://github.com/advisories/GHSA-j8xg-fqg3-53r7

Change-Id: I7c09c8ea4b978d7901404a9b431b8839f8772a6e

Commit:
build: Updating npm dependencies

* eslint-config-wikimedia: 0.25.0 → 0.26.0
* grunt-eslint: 24.0.1 → 24.3.0

Change-Id: I77756800c871651ce9118fa794b112d14e00d95e

Commit:
build: Updating dependencies

composer:
* mediawiki/mediawiki-phan-config: 0.12.1 → 0.14.0

npm:
* grunt-banana-checker: 0.11.0 → 0.11.1

Change-Id: I17350c164c1bba9a78bb358bc82b07041fc7c46f

Commit:
build: Updating mediawiki/mediawiki-codesniffer to 43.0.0

Change-Id: Ia01f84bca1c56a73501cfc01dcef6880de77a049

Commit:
build: Updating dependencies

composer:
* mediawiki/mediawiki-codesniffer: 43.0.0 → 44.0.0
* mediawiki/minus-x: 1.1.1 → 1.1.3
* php-parallel-lint/php-parallel-lint: 1.3.2 → 1.4.0

npm:
* eslint-config-wikimedia: 0.26.0 → 0.28.2
* grunt-banana-checker: 0.11.1 → 0.13.0
* braces: 3.0.2 → 3.0.3
  * https://github.com/advisories/GHSA-grv7-fg5c-xmjg
* micromatch: 4.0.5 → 4.0.8
  * https://github.com/advisories/GHSA-952p-6rrq-rcjv

Change-Id: I8e0449987f7621f3779b256473c6cefe9d3d6aee

Commit:
build: Updating mediawiki/mediawiki-codesniffer to 45.0.0

Change-Id: Id2d5c8017904a4ae1f334a9dd86563ef794f4185

jdforrester committed:
tests: Remove use of ReflectionMethod::setAccessible(), no-op

As well as being a no-op now that we're in PHP 8.1+, these are
deprecated and will trigger errors in PHP 8.5.

Bug: T406744
Change-Id: I42e7c43dfa0dd1002c9ecc3f411d21d22d695d33

umherirrender committed:
Use namespaced WikiMap class

Requires MediaWiki 1.40, needed to work with 1.44+ (db47e7f715)

Change-Id: Ie4e230a3c89f626e14afea8e77b494e02ddfdf7b

DanielButAtWork commented:
Yeah we are phasing out our usage of Sentry, closing this
